src/providers/theme/ThemeProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import {
+  CssBaseline,
+  ThemeProvider as MUIThemeProvider,
+  PaletteMode,
+  createTheme,
+} from '@mui/material';
+import { ReactNode, useMemo, useState } from 'react';
+import { ThemeContext, ThemeContextProps } from './ThemeContext';
+import { getTheme } from './theme';
+
+type ThemeProviderProps = {
+  children: ReactNode;
+};
+
+export const ThemeProvider = ({ children }: ThemeProviderProps): JSX.Element => {
+  const [mode, setMode] = useState<PaletteMode>('light');
+
+  const themeContextValue = useMemo<ThemeContextProps>(
+    () => ({
+      toggleTheme: () => {
+        setMode((prevMode: PaletteMode) => (prevMode === 'light' ? 'dark' : 'light'));
```

Comment:
Is it possible to use typescript enum instead of strings?

Reply:
It is strongly typed, PaletteMode defines literal type 'light' | 'dark' so it is determined to set only one of them. I cannot see a need to create enum

Reply:
I checked that and it is possible if you create an enum like this:
```
enum ThemeMode {
  light = 'light',
  dark = 'dark',
}
```
It would look better in the code if we used enum